src/components/layertree/LayerTree.js

```diff
@@ -217,42 +207,38 @@ class LayerTree extends Component {
         tabIndex={tabIndex}
         className={classNameToggle}
         onClick={() => {
-          if (!level || !LayerTree.hasChildren(item)) {
-            LayerTree.onInputClick(item);
+          if (layer.getChildren().length === 0) {
+            layer.setVisible(!layer.getVisible());
```

Comment:
That's why onInputClick was for , to avoid duplicating this code, so you don't have to test it on each case.

Reply:
That makes sense. Sorry, I removed it for testing and forgot to re-add it.